core/src/main/java/org/apache/iceberg/rest/responses/CreateNamespaceResponse.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.responses;
+
+import java.util.Map;
+import java.util.Objects;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+/**
+ * Represents a REST response for a request to create a namespace / database.
+ */
+public class CreateNamespaceResponse {
+
+  private Namespace namespace;
+  private Map<String, String> properties;
+
+  public CreateNamespaceResponse() {
+    // Required for Jackson deserialization.
+  }
+
+  private CreateNamespaceResponse(Namespace namespace, Map<String, String> properties) {
+    this.namespace = namespace;
+    this.properties = properties;
+    validate();
+  }
+
+  CreateNamespaceResponse validate() {
+    Preconditions.checkArgument(
+        namespace != null && !namespace.isEmpty(),
+        "Cannot build a response to create a namespace without specifying the namespace");
```

Comment:
These messages are okay, but I typically prefer a more direct statement that is simpler to parse. This says "build" when this is really a validation and says "without specifying the namespace", which is a long way to say it is either null or empty. It is also inaccurate for empty namespaces because the namespace was included, it just makes no sense.

I think the best way to fix this is to remove the empty check and just check for null, with a simple error message: "Invalid namespace: null".

Reply:
Yeah that sounds good to me. And the empty namespace should probably be allowed, in case people want to model something using it (or there are situations where it's safe to omit... which would probably depend on the server implementation but isn't something we should disallow).

Reply:
Updated to simply check for null and use the simplified language.